lib/safe-http-client.js

```diff
@@ -0,0 +1,134 @@
+import dns from 'node:dns/promises'
+
+import {fetch} from 'undici'
+import ipaddr from 'ipaddr.js'
+
+import {IMAGE_MIME_TYPES} from './constants.js'
```

Comment:
```suggestion
import dns from 'node:dns/promises'
import ipaddr from 'ipaddr.js'
import {fetch} from 'undici'
import {IMAGE_MIME_TYPES} from './constants.js'
```

Sorting, no blank lines.

Q: As Node 16 is not maintained, we can go with Node 18 as a base here, which supports `fetch`. I wouldn’t do that for the unified plugins / utilities as half of them is just released with Node 16 support, so having mixed 16 and 18 minimum is complex, but as this is a new project intended for a very different place, we could do it here?

Reply:
I needed `undici` anyway because of `MockAgent` in tests and since the package is there, might as well use the `fetch` export and not rely on a experimental Node >18 only feature. I can remove it, but it comes for free at this point.